Remotion/Web/Core/UI/Controls/ILazyValidator.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Web.UI;
+
+namespace Remotion.Web.UI.Controls
+{
+  public interface ILazyValidator: IValidator
+  {
+    /// <summary>Performs validation on the already validated associated input control again and updates the <see cref="P:System.Web.UI.WebControls.BaseValidator.IsValid" />
+    /// property if there are no more validation failures.</summary>
+    void LazyValidate ();
```

Comment:
I think we need to talk about the name of the interface and its method with @MichaelKetting. As I understand it, we do not want to change the result of the validation but only set the appropriate `ErrorMessage`. `ComputeErrorMessage` or something feels more correct than `LazyValidate` since we don't want to validate again only compute the correct error message.

Reply:
@a-ctor correct: "lazy" ist just a temp name, the actual api name is not determined yet. 
@Frazerus: I'm happy to start the discussion with a couple of preexisting ideas :)

Reply:
I think we should have this as a full validation iteration, as we would not need to render the control if we don't have unhandled validation errors & no errors on the boclist itself. 

There is IMO no reason to not also set IsValid if we are changing the ErrorMessage, given that we could find that there are no errors anymore.

Reply:
@Frazerus option 2 because the validation is not actually lazy.
* IValidatorWithDynamicErrorMessage -> RefreshErrorMessage
* IValidatorWithLazyErrorMessage -> EnsureErrorMessageEvaluated

which one fits better?

Reply:
> I think we should have this as a full validation iteration, as we would not need to render the control if we don't have unhandled validation errors & no errors on the boclist itself.
> There is IMO no reason to not also set IsValid if we are changing the ErrorMessage, given that we could find that there are no errors anymore.

Since we need the information "BocList has validation errors somewhere" anyway, it's best to just get the list of left over validation errors during render but not change the IsValid status. If the list is empty at that time, we provide the default error message instead of a list of specific errors. This has the added advantage, that it provides a consistent experice for screenreader users (they know on the top level that there's validation errors in this BocList. For seeing users, this one validation error at the bottom of the boclist isn't a problem because there's likely much more space used elsewhere by BocList.